src/routes/tools/_common/_helpers.js

```diff
@@ -0,0 +1,152 @@
+import { rollup, sort, group } from "d3-array";
+
+/**
+ * Groups data for 2 or more timeseries by year, outputs a single timeseries with
+ * min and max value for each year.
+ * Used for creating envelopes to represent an ensemble range.
+ * This data structure is used as an input to the d3 area function
+ * @param {array} _data
+ * @return {object}
+ */
+export function buildEnvelope(_data) {
+  const dataMap = rollup(
+    _data,
+    (v) => v.map((i) => i.value),
+    (d) => d.date.getFullYear()
+  );
+  const dataArr = Array.from(dataMap);
+  return dataArr.map(([key, value]) => {
+    const sortedArr = sort(value);
+    return {
+      date: new Date(key, 11, 31),
+      min: sortedArr[0],
+      max: sortedArr[1],
+    };
+  });
+}
+
+/**
+ * Flattens an array of series objects of the type:
+ * {
+ *  id: HadGEM2-ES,
+ *  values:[
+ *    { date: Date Sun Dec 31 1950, value: 90},
+ *    { date: Date Sun Dec 31 1951, value: 95},
+ *    { date: Date Sun Dec 31 1952, value: 97},
+ *    ...
+ *  ]} 
+ * to:
+   [
+ *   { date: Date Sun Dec 31 1950, value: 90, year: 1950, id: HadGEM2-ES, label: HadGEM2-ES (Warm/Dry)},
+ *   { date: Date Sun Dec 31 1951, value: 95, year: 1950, id: HadGEM2-ES, label: HadGEM2-ES (Warm/Dry)},
+ *   { date: Date Sun Dec 31 1952, value: 97, year: 1950, id: HadGEM2-ES, label: HadGEM2-ES (Warm/Dry)},
+ *   ...
+ * ] 
+ * Used for chart tooltips
+ * @param {array} _data - array of series objects
+ * @return {array}
+ */
+export function flattenData(_data) {
+  return _data.reduce((acc, series) => {
```

Comment:
feel free to take it or leave it, but you could also just do 

```js
data.series.map(({ date, value }) => ({ 
   ...d,
  year: +d.date.getFullYear(),
  id: data.series.id,
  label: data.series.label,
}))
```

I mention it because it's generally good to avoid using `reduce` when possible as it can make the code more difficult to follow.